msal/tests/Test.MSAL.NET.Unit/AuthorityAliasesTests.cs

```diff
@@ -0,0 +1,202 @@
+            {
+                HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler()
+                {
+                    Url = string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/oauth2/v2.0/token",
+                        TestConstants.ProdPrefNetworkEnv, TestConstants.Utid),
+                    Method = HttpMethod.Post,
+                    PostData = new Dictionary<string, string>()
+                    {
+                        {"grant_type", "refresh_token"}
+                    },
+                    // return not retriable status code
+                    ResponseMessage = MockHelpers.CreateInvalidGrantTokenResponseMessage()
+                });
+
+                try
+                {
+                    result = null;
+                    result = app.AcquireTokenSilentAsync(TestConstants.ScopeForAnotherResource,
+                        app.GetAccountsAsync().Result.First(),
+                        string.Format(CultureInfo.InvariantCulture, "https://{0}/{1}/", envAlias, TestConstants.Utid),
+                        false).Result;
+                }
+                catch (AggregateException ex)
+                {
+                    Assert.IsNotNull(ex.InnerException);
+                    Assert.IsTrue(ex.InnerException is MsalUiRequiredException);
+                }
+
+                Assert.IsNull(result);
+            }
+        }
+
+        private void ValidateCacheEntitiesEnvironment(TokenCache cache, string expectedEnvironment)
+        {
+            var accessTokens = cache.GetAllAccessTokensForClient(new RequestContext(new MsalLogger(Guid.NewGuid(), null)));
+            foreach (var at in accessTokens)
+            {
+                Assert.AreEqual(expectedEnvironment, at.Environment);
+            }
+
+            var refreshTokens = cache.GetAllRefreshTokensForClient(new RequestContext(new MsalLogger(Guid.NewGuid(), null)));
+            foreach (var rt in refreshTokens)
+            {
+                Assert.AreEqual(expectedEnvironment, rt.Environment);
+            }
+
+            var idTokens = cache.GetAllIdTokensForClient(new RequestContext(new MsalLogger(Guid.NewGuid(), null)));
+            foreach (var id in idTokens)
+            {
+                Assert.AreEqual(expectedEnvironment, id.Environment);
+            }
+
+            var accounts = cache.GetAllAccounts(new RequestContext(new MsalLogger(Guid.NewGuid(), null)));
+            foreach (var account in accounts)
+            {
+                Assert.AreEqual(expectedEnvironment, account.Environment);
+            }
+
+            IDictionary<AdalTokenCacheKey, AdalResultWrapper> adalCache =
+                AdalCacheOperations.Deserialize(cache.legacyCachePersistance.LoadCache());
+
+            foreach (KeyValuePair<AdalTokenCacheKey, AdalResultWrapper> kvp in adalCache)
+            {
+                Assert.AreEqual(expectedEnvironment, new Uri(kvp.Key.Authority).Host);
```

Comment:
>new Uri(kvp.Key.Authority).Host [](start = 53, length = 31)

nit: would be great if it was not required to new up an object.

Reply:
i need to extract host form url string here